sftp/client.go

```diff
@@ -0,0 +1,104 @@
+//go:generate mockgen -destination=mock_sftp/mock_sftp_client.go -package mock_sftp github.com/rudderlabs/rudder-go-kit/sftp SFTPClient
+package sftp
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/pkg/sftp"
+	"golang.org/x/crypto/ssh"
+)
+
+// SSHConfig represents the configuration for SSH connection
+type SSHConfig struct {
+	Host       string
+	Port       int
+	User       string
+	AuthMethod string
+	PrivateKey string
+	Password   string // Password for password-based authentication
+}
+
+func ConfigureSSHClient(config *SSHConfig) (*ssh.ClientConfig, error) {
+	if config == nil {
+		return nil, fmt.Errorf("config should not be nil")
+	}
+
+	if config.Host == "" {
+		return nil, fmt.Errorf("host should not be empty")
+	}
+
+	if config.User == "" {
+		return nil, fmt.Errorf("user should not be empty")
+	}
+
+	var authMethods []ssh.AuthMethod
+
+	switch config.AuthMethod {
+	case PasswordAuth:
+		authMethods = []ssh.AuthMethod{ssh.Password(config.Password)}
+	case KeyAuth:
+		privateKey, err := ssh.ParsePrivateKey([]byte(config.PrivateKey))
+		if err != nil {
+			return nil, err
```

Comment:
Should we add more context in here?
```suggestion
			return nil, fmt.Errorf("cannot parse private key: %w", err)
```